seismostats/analysis/bvalue/base.py

```diff
@@ -214,6 +216,36 @@ def func(sample):
         self.idx = temp_idx
         return std
 
+    def p_lilliefors(self, n=100):
+        '''
+        p-value of the Lilliefors test. Weights are not yet implemented.
+        Magnitudes are dithered to continuous ones by resampling the
+        distribution of the binned magnitudes, taking into account the
+        exponential distribution of the magnitudes (in contrast to
+        Herrmann et al, 2020).
+        Source:
+        - Herrmann, M. and W. Marzocchi (2020). "Inconsistencies and Lurking
+        Pitfalls in the Magnitude-Frequency Distribution of High-Resolution
+        Earthquake Catalogs".
+        Seismological Research Letters 92(2A). doi: 10.1785/0220200337
+        - Lilliefors, Hubert W. "On the Kolmogorov-Smirnov test for the
+        exponential distribution with mean unknown." Journal of the American
+        Statistical Association 64.325 (1969): 387-389.
+        '''
+        self.__is_estimated()
```

Comment:
it can be slightly confusing that you have to calculate a b value before you can do a lilliefors test, since the result does not depend on the calculated b-value. I understand that it does depend on the side products of the calculation.. and I don't know what I would change about it, other than explaining this in the documentation

Reply:
although, it does depend on the b-value for the dithering. in any case, I would explain in the docstring that this p-value does not depend on the calculated b-value (except for dithering) and that the p-value is produced by creating many ditherings and does therefore not yield a deterministic result. I got quite some variation when using a large magnitude sample but 100 dithering realizations..

Reply:
We can also set the number higher than 100 in order to reduce the fluctuation..

Reply:
hmm yeah but then it will just take longer, and there is no n that guarantees that fluctuations will be below some limit, since it always depends on the sample